### PR TITLE
20240509 @mrwatson-de v1.1

### PR DESCRIPTION
**Pull request by mrwatson-de**
- Improve to not need MBS plugin
- Use `json_data_stored_in_serial_field` to store all the data you need

Hi Nils,

nice idea!

I've tweaked your file to use a *native* technique, where we can store (a lot of) data in a serial number field rather than in the (cramped) default field.

Note that i have ADDED a new field to demonstrate this. This leaves the old default field redundantly laying around, so you may want to either just delete the old default field in my version of the file, or build the changes into your original file REPLACING the old default field.

Happy FileMaking

MrWatson

### CODE REVIEW

**Comment by mrwatson-de** [comment]
Here is a little more about the changes

# The technique

So, I'm storing the data in the serial field's next value:

Serial number field next value


Note that an extra zzz element is required in the JSON to protect your data from being incremented. Adding a dummy number at the far right of the JSON string means this dummy number is incremented when a new record is created, leaving the rest of the json data intact:

zzz element

This works, because 

- JSON elements are stored alphabetically, so the name `zzz`causes this element to be the last in the list, and thus at the far right of the string
- the GetNextSerialValue function increments only the first number it encounters from the right hand end of the string, leaving the rest of the string intact.


# How much data can be stored?

As far as the question goes of how much data can be stored…I experimented with a large array…

test-how big can this thing be

…and, although I didn't test the *limit*,  the simple answer is, we'll be able to store as much info as we need!

test-how big can this thing be-BIG

In other words this technique could be used not only for version & build info, but *could* be extended to include all kinds of initialisation info.

**Comment by mrwatson-de** [comment]
Changes made:

In Script "saveBuildinfo()"

- Add Set Next Serial Value
- Change to serial number field
- Remove MBS + SQL stuff
- Remove unnecessary Commit + Pause
- Add history comment

In Script  "onFirstWindowOpenTrigger()"…

- Change to serial number field
- Remove MBS stuff
- Add history comment

In Layout

- Remove MBS stuff

**Comment by nilswald** [comment]
Wow, excellent improvement! We could simply add the number after the json object an remove it before processing the payload. The value does not have to be valid json: {"build":102,"timestamp":63850928014,"version":[1,0,0]}-3  removing everything after "}-"

**Comment by nilswald** [comment]
and btw, 1MB payload: check 👍

chatgpt
